Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PaymentMethodTokenizationViewModel.swift

```diff
@@ -309,54 +309,260 @@ class PaymentMethodTokenizationViewModel: NSObject, PaymentMethodTokenizationVie
     func handleFailedTokenizationFlow(error: Error) {
         Primer.shared.primerRootVC?.handle(error: error)
     }
-    
 }
 
 extension PaymentMethodTokenizationViewModel {
-    func handle(error: Error) {
-        assert(true, "\(self.self).\(#function) should be overriden")
+    
+    internal func handleErrorBasedOnSDKSettings(_ error: Error, isOnResumeFlow: Bool = false) {
+        
+        let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
+        if settings.isManualPaymentHandlingEnabled {
+            if isOnResumeFlow {
+                PrimerDelegateProxy.onResumeError(error)
+            }
+            handle(error: error)
+        } else {
+            PrimerDelegateProxy.primerDidFailWithError(error, data: nil, decisionHandler: { errorDecision in
+                if let errorMessage = errorDecision?.additionalInfo?[.message] as? String {
+                    let merchantError = PrimerError.merchantError(message: errorMessage, userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"])
+                    self.handle(error: merchantError)
+                } else {
+                    self.handle(error: NSError.emptyDescriptionError)
+                }
+            })
+        }
     }
+}
+
+extension PaymentMethodTokenizationViewModel {
     
-    func handle(newClientToken clientToken: String) {
-        assert(true, "\(self.self).\(#function) should be overriden")
+    @objc func executeCompletionAndNullifyAfter(error: Error? = nil) {
+        self.completion?(nil, error)
+        self.completion = nil
     }
     
-    func handleSuccess() {
-        assert(true, "\(self.self).\(#function) should be overriden")
+    func validateReturningPromise() -> Promise<Void> {
+        return Promise { seal in
+            do {
+                try self.validate()
+                seal.fulfill()
+            } catch {
+                seal.reject(error)
+            }
+        }
     }
 }
 
-private extension UIColor {
-    var hex: String? {
-        var red: CGFloat = 0
-        var green: CGFloat = 0
-        var blue: CGFloat = 0
-        var alpha: CGFloat = 0
+extension PaymentMethodTokenizationViewModel {
+    
+    internal func handleContinuePaymentFlowWithPaymentMethod(_ paymentMethod: PaymentMethodToken) {
+                
+        let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
+        
+        if Primer.shared.flow.internalSessionFlow.vaulted {
+            PrimerDelegateProxy.tokenAddedToVault(paymentMethod)
+            self.handleSuccess()
+        } else if settings.isManualPaymentHandlingEnabled {
+            
+            PrimerDelegateProxy.onTokenizeSuccess(paymentMethod, resumeHandler: self)
+            PrimerDelegateProxy.onTokenizeSuccess(paymentMethod, { err in
+                if let err = err {
+                    self.handleFailedTokenizationFlow(error: err)
```

Comment:
I believe here you should call the `completion` and end where all other flows would end.